### PR TITLE
New: Added filter and sort options to Collections

### LINKED ISSUE #7338: Filtering for collections screen
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

On the collections screen, a couple more filters would be handy to have.

### Describe the solution you'd like

1) Filter for collections missing 1+ movies.
2) Sorting by number of missing movies.
3) Filter for complete collections.

### Describe alternatives you've considered

None

### Anything else?

N/A

[AB#3233](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/3233)

### PR DESCRIPTION
**Pull request by ricci2511**
#### Database Migration
NO

#### Description
Added 2 new filter options to filter by missing or complete collections, and a sort option to sort by missing movies in collections.

#### Screenshot (if UI related)
filters

sort

#### Issues Fixed or Closed by this PR

* Closes #7338

### CODE REVIEW

File: src/Radarr.Api.V3/Collections/CollectionController.cs [hunk 1]
```diff
@@ -145,6 +145,11 @@ private IEnumerable<CollectionResource> MapToResource(List<MovieCollection> coll
                     var movieResource = movie.ToResource();
                     movieResource.Folder = _fileNameBuilder.GetMovieFolder(new Movie { MovieMetadata = movie }, namingConfig);
 
+                    if (_movieService.FindByTmdbId(movie.TmdbId) == null)
```
Comment (Qstick) [inline src/Radarr.Api.V3/Collections/CollectionController.cs:148]: This is going to result in thousands of db calls for a collection endpoint call. The better way is to make a single call to get db movies with collections and cross compare to that list.

**Review by Qstick — CHANGES_REQUESTED** [review]

File: src/Radarr.Api.V3/Collections/CollectionController.cs (hunk 1 above)
Comment (ricci2511) [reply src/Radarr.Api.V3/Collections/CollectionController.cs:148]: Yes, my bad, will take a look into it and do another commit, thanks.

**Comment by ricci2511** [comment]
> I would revert anything related to coding style changes. 

Will do, i just ran `yarn lint --fix` since i thought it was recommended, but it does make it harder to see at first glance what i actually changed for this feature.

**Comment by ricci2511** [comment]
Sorry for the previous mess :(

But I think it should be fine now, reverted the code styling changes and reduced database calls like @Qstick suggested.

**Comment by Qstick** [comment]
No problem, looking better. 

Could possibly refine it a bit more by adding a method to movie service and repo to grab movies with collections direct with db query instead of calling `all` then filtering down. Much less impact than previous change but may help shave time off the massive DBs out there (100k movies +)

**Comment by ricci2511** [comment]
> No problem, looking better.
> 
> Could possibly refine it a bit more by adding a method to movie service and repo to grab movies with collections direct with db query instead of calling `all` then filtering down. Much less impact than previous change but may help shave time off the massive DBs out there (100k movies +)

Alright, I don't have that much experience with C# but I'll give it a try anyways later today when I have time.

**Comment by Qstick** [comment]
Thanks, feel free to pop by discord if you need any help. Thanks for jumping in on this

**Review by Qstick — APPROVED** [review]

**Pull request merged**